### ISSUE TITLE
Importing Private Key asks for another Invitation code

### ISSUE THREAD

**Issue by FioX0** [planetarium/9c-launcher#984]
Hi Team,

When importing a private key instead of creating a new account. It asks for an invitation code to insert this existing private key into the launcher. The only workaround that exists for this is to create an extra account and hunt for another invitation code, which is not really great.

This can by bypassed if the user in question still has access to the 9c files on a previous PC or take a copy of their keyfile before formatting their PC. If you place the keyfile in the new installation (or new pc) then the launcher will read the data from the keyfile and won't ask for a new invitation code. So at this current time, using the keyfile as a recovery method is the only viable option that doesn't require creating a new twitter account and attempt to hunt for a new invitation code.

**Comment by BasixKOR** [comment]
Potentially duplicate of #893?

**Comment by FioX0** [comment]
Still happening in current version.

**Comment by BasixKOR** [comment]
Sorry for the inconvenience. Could you check the issue again in the v100087 release? We have landed a fix for the similar case and I'm guessing that fix should work on this case too.

**Comment by FioX0** [comment]
Still happening for users I'm afraid. We had 2 people mention it on discord and had to create new onboarding accounts to get invitation codes.

**Comment by FioX0** [comment]
Just for your information this is still happening in v97, almost everyday we have a few people having this issue.

**Comment by Akamig** [comment]
...A year past, and I just tested this and the result was,

- No file in keystore folder, import activated private key string, login -> Login Success, Game Started
- Other File in keystore folder, forgot password, import activated private key string, login -> Login Success, Game Started
- No file in keystore folder, import non-activated private key string, login -> MissingActivation
- Other File in keystore folder, forgot password, import non-activated private key string, login -> MissingActivation

...aand we also substituted MissingActivation view with just Activation view, but simply put this shouldn't be happening for already activated private key now. and I think we can close this issue. can I get a confirmation?
